feat(indemnite-licenciement): ajout de l'outil de rupture conventionnelle

m-maillot opened the pull request:
closes [#5626](https://github.com/SocialGouv/code-du-travail-numerique/issues/5626)

m-maillot committed:
start

m-maillot committed:
feat(indemnite-licenciement): use calculate from information step

m-maillot committed:
feat(indemnite-licenciement): gestion des ANI avec plusieurs types de rupture

m-maillot committed:
feat(indemnite-licenciement): mise en place du frontend

m-maillot committed:
feat(indemnite-licenciement): changement du wording pour la rupture conventionnelle

m-maillot committed:
fix(indemnite-licenciement): modification du wording

m-maillot committed:
fix(indemnite-licenciement): fix unit test

m-maillot committed:
fix(indemnite-licenciement): fix build

m-maillot committed:
feat(indemnite-licenciement): gestion de l'ineligibilité

m-maillot committed:
feat(indemnite-licenciement): changement du wording pour la rupture conventionnelle

m-maillot committed:
Merge branch 'dev' into feat/modeles-rupture

m-maillot committed:
fix: unit test + lint

m-maillot committed:
fix: unit test

m-maillot committed:
fix: unit test

m-maillot committed:
fix: unit test

m-maillot committed:
fix: import optimization

carolineBda commented on packages/code-du-travail-modeles/src/publicodes/RuptureConventionnellePublicodes.ts:
```diff
@@ -85,22 +70,82 @@ class RuptureConventionnellePublicodes
+      return this.calculateSituation(args, targetRule);
+    } else {
+      const situations = reasons.map(({ name, rule, value }) => {
+        console.log(`Compute for reason: ${name}`);
```
```suggestion
```

carolineBda committed:
empty commit

m-maillot committed:
Update packages/code-du-travail-modeles/src/publicodes/RuptureConventionnellePublicodes.ts

Co-authored-by: Caroline <4971715+carolineBda@users.noreply.github.com>

m-maillot committed:
feat: suppression de la question sur l'inaptitude

m-maillot committed:
feat: merge des missing args

carolineBda committed:
Merge branch 'dev' into linked/feat/add-rupture-co

carolineBda reviewed (changes requested):
![Screenshot from 2024-04-02 13-33-32](https://github.com/SocialGouv/code-du-travail-numerique/assets/4971715/c4a1d649-762b-4ebc-a6e3-241e3bceadf7)

Regression sur la page résultat de l'IL on dit " l’indemnité de rupture conventionnelle "

carolineBda committed:
Merge branch 'dev' into linked/feat/add-rupture-co

dependabot[bot] committed:
chore(deps): bump express from 4.18.2 to 4.19.2 (#5717)

Bumps [express](https://github.com/expressjs/express) from 4.18.2 to 4.19.2.
- [Release notes](https://github.com/expressjs/express/releases)
- [Changelog](https://github.com/expressjs/express/blob/master/History.md)
- [Commits](https://github.com/expressjs/express/compare/4.18.2...4.19.2)

---
updated-dependencies:
- dependency-name: express
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

m-maillot committed:
fix: mauvais texte pour afficher le résultat de l'indemnite de licenciement

m-maillot commented:
> ![Screenshot from 2024-04-02 13-33-32](https://private-user-images.githubusercontent.com/4971715/318763062-c4a1d649-762b-4ebc-a6e3-241e3bceadf7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxMzU1MjUsIm5iZiI6MTcxMjEzNTIyNSwicGF0aCI6Ii80OTcxNzE1LzMxODc2MzA2Mi1jNGExZDY0OS03NjJiLTRlYmMtYTZlMy0yNDFlM2JjZWFkZjcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwMyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDNUMDkwNzA1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjQ2M2U2NzJmMjQ2MGU4MDA2YTE2OWMwOWY0MjM3MTAyMzYwODA3NzZmNDI4MWExMmI0OTg0ZGZhNjBkY2U2NiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.81xq8PMhlQYXZ1tXanPYY4K6mjaHrOaDBj38wgUNhQ8)
> 
> Regression sur la page résultat de l'IL on dit " l’indemnité de rupture conventionnelle "

C'est corrigé :)

maxgfr commented on packages/code-du-travail-frontend/pages/outils/index.tsx:
```diff
@@ -89,9 +89,10 @@ export async function getStaticProps() {
     } else {
       result = await getToolsByIdsAndSlugs();
     }
+    const isProd = process.env.NEXT_PUBLIC_SENTRY_ENV === "production"
```
Je  pense qu'il est plus preferrable d'utiliser : `NEXT_PUBLIC_IS_PRODUCTION_DEPLOYMENT`, que de se baser sur la variable de sentry

m-maillot replied:
Mais oui ! Comme je suis passé à coté de ça 🤪

maxgfr commented on packages/code-du-travail-frontend/src/outils/CommonIndemniteDepart/steps/Resultat/components/DecryptResult.tsx:
```diff
@@ -6,21 +6,23 @@ type Props = {
   isAgreementSupported: boolean;
   legalResult: string;
   agreementResult?: string;
+  label: string;
```
Je pense on aurait pu le typer fort en `licenciement` | `rupture`

```suggestion
  label: 'licenciement' | 'rupture conventionnelle'
```

maxgfr replied:
Ou alors reprendre le type : `IndemniteDepartType` que t'utilises à d'autre endroit

maxgfr commented on packages/code-du-travail-frontend/src/outils/CommonIndemniteDepart/steps/Resultat/components/FilledElements.tsx:
```diff
@@ -41,39 +46,19 @@ export default function FilledElements(props: Props) {
+              return (
+                <li key={item.text}>
+                  {item.text}&nbsp;:&nbsp;{item.value}
+                  {item.detail && "*"}
+                  {item.detail && (
+                    <Paragraph italic noMargin>
+                      * {item.detail}
```
Je pense qu'il y aura deux `*` si on fait çaAvant, on affichait la deuxieme étoile que si y'avait pas de `isAgreementBetter`

m-maillot replied:
Ha oui ! Bien vu... ça ne va pas m'aider ça 😅

m-maillot replied:
En fait, je le gère dans les composants de chaque simu. Mais il y avait bien un bug car j'avais inversé une condition :)

maxgfr commented on packages/code-du-travail-frontend/src/outils/CommonIndemniteDepart/steps/Resultat/components/ForMoreInfo.tsx:
```diff
@@ -2,27 +2,17 @@ import Link from "next/link";
-      </p>
+      {article}
       <Paragraph italic fontSize="small">
         {message}
```
On aurait peut être pu garder ce composant en le renommer `ForMoreInfoIndemniteLicenciement`, car de la sorte, il est pas si utile que ça. Le but de ce composant était de garder la logique métier d'affichage

m-maillot replied:
Yep, je supprime le composant qui n'a plus lieu d'exister. Chaque simu implémente les infos qu'il a besoin et la rupture a besoin de beaucoup moins d'info à ce niveau là.

maxgfr commented on packages/code-du-travail-frontend/src/outils/CommonIndemniteDepart/steps/Salaires/components/TempsPartiel.tsx:
```diff
@@ -1,17 +1,25 @@
-        Le calcul de l’indemnité de licenciement dans le cas d’une alternance de
-        temps plein et de temps partiel est actuellement en cours de
-        développement.
+        Le calcul de l’indemnité de{" "}
+        {type === IndemniteDepartType.LICENCIEMENT
+          ? "licenciement"
+          : "rupture conventionnelle"}{" "}
+        dans le cas d’une alternance de temps plein et de temps partiel est
+        actuellement en cours de développement.
```
On utilise souvent cette logique, il faudrait voir comment la mettre ailleurs, c'est ce que je disais dans une des suggestion que j'avais fait plus haut

m-maillot replied:
Je serais partant de le faire dans un second refacto. Car j'ai peur aussi que le métier demande plus de modifications sur les textes et qu'en fait on doit revoir nos composants plus globalement. Ca te va ?

maxgfr commented on packages/code-du-travail-frontend/src/outils/CommonSteps/Agreement/store/store.ts:
```diff
@@ -173,15 +168,15 @@ const createCommonAgreementStore: StoreSlicePublicode<
     },
     onEnterpriseSearch: (data) => {
       matopush([
         MatomoBaseEvent.TRACK_EVENT,
         MatomoSearchAgreementCategory.ENTERPRISE_SEARCH,
-        toolName,
+        type,
         JSON.stringify(data),
       ]);
```
c'est top, du coup, ça règle la problématique dont on parlait pour la différenciation des events entre la rupture co et l'indemnité de licenciement

maxgfr replied:
Car en soit, le toolName et le type sont assez similaire. Pourquoi avoir rename la variable ?

maxgfr replied:
D'ailleurs, pour rester sur cette problématique des events, j'aimerai bien voir comment on pourrait refacto pour l'enlever de la logique des store, et que ça soit géré dans une classe à part entière. Je trouve ça tellement pas top de mettre la logique dans le store en fait

Viczei commented on packages/code-du-travail-frontend/src/outils/CommonIndemniteDepart/steps/ContratTravail/store/store.ts:
```diff
@@ -64,10 +67,11 @@ const createContratTravailStore: StoreSlice<
-              PublicodesSimulator.INDEMNITE_LICENCIEMENT
-            );
+          const simulator =
+            type === IndemniteDepartType.LICENCIEMENT
+              ? PublicodesSimulator.INDEMNITE_LICENCIEMENT
+              : PublicodesSimulator.RUPTURE_CONVENTIONNELLE;
```
C'est dommage de ne pas avoir un seul enum

m-maillot replied:
C'est assez logique ici. Le `PublicodesSimulator` est situé dans le modèle. Il liste l'ensemble des simulateurs supportés par nos modèles (Préavis de retraitre, indemnité licenciement, rupture co)
Le `IndemniteDepartType` quand à lui permet de paramétrer la base commune entre Indemnité licenciement et Rupture co. 
Il est donc compliqué d'en faire qu'un seul.

maxgfr commented on packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/steps/Resultat/Eligible.tsx:
```diff
@@ -1,27 +1,30 @@
+  useIndemniteDepartStore,
+} from "../../../CommonIndemniteDepart/store";
+import { getResultMessage } from "./utils";
+// Do not optimize the following import
```
Il faudrait voir comment on peut régler ce soucis, c'est vraiment atroce 😢

A l'époque, j'avais essayé d'upgrade typescript, etc. Rien ne bougeait.

D'ailleurs dans le futur, il faudrait qu'on se fasse une tâche upgrade typescript + next.js

maxgfr replied:
Même si malheuresement je ne pense pas que ça règle grand chose

m-maillot replied:
Je n'ai pas regardé plus loin, on avait déjà regardé et on avait fait chou blanc... mais c'est vrai que c'est fastidieux !

maxgfr commented on packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/steps/Resultat/Eligible.tsx:
```diff
@@ -110,17 +113,45 @@ export default function Eligible() {
+              text: "Arrêt de travail au moment du licenciement",
+              value: arretTravail === "oui" ? "Oui" : "Non",
+            },
+          ].concat(
```
J'aime bien cette approche, car ça permet au composant d'être assez générique entre les deux simulateurs ! Bien joué

maxgfr commented on packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/steps/Resultat/index.tsx:
```diff
@@ -1,37 +1,15 @@
-    return <ErrorPublicodes />;
-  }
-
-  return <>{isEligible ? <Eligible /> : <Ineligible />}</>;
+  return (
+    <StepResultat
+      eligibleComponent={<Eligible />}
+      ineligibleComponent={<Ineligible />}
+    />
+  );
```
A l'époque, on avait gardé l'`init`, car ça permettait de re-update le résultat manuellement et le calculer si des éléments de la situation avait bougé. Je suis pas sûr d'avoir bien compris le mécanisme qui permet de gérer ce use-case

m-maillot replied:
Il est toujours là mais un cran plus haut : https://github.com/SocialGouv/code-du-travail-numerique/pull/5703/files#diff-cc75f0b01797185afb8cea6c219473d616b5aa32a562b1703330bbc5496655cdR21

J'ai soustrait une partie du composant initial pour le rendre customisable pour chaque simulateur.

maxgfr commented on packages/code-du-travail-frontend/src/outils/IndemniteLicenciement/index.tsx:
```diff
@@ -1,191 +1,78 @@
+      displayTitle={displayTitle}
+      tool={IndemniteDepartType.LICENCIEMENT}
+      steps={steps}
+    />
```
C'est bien mieux comme ça 💪

maxgfr commented on packages/code-du-travail-frontend/src/outils/RuptureCoventionnelle/steps/Anciennete/index.tsx:
Perso ça me choque pas de dupliquer l'information, car en soit d'un point de vue métier, il y a quand même une approche différente. 

La problématique est que lorsqu'il faudra modifier un texte, il faudra penser aux deux occurences (sur l'indemnité et la rupture)

m-maillot replied:
Les deux nouveaux ancienneté partagent pas mal leurs composants. On pourrait en partager encore plus. Mais sur les quelques questions de cette écran, les dates changent car on ne demande pas la même chose (label) et on en demande une en moins. La partie en commun est celle sur les absences où j'ai utilisé le composant commun pour la liste des absences. Je pense qu'il est préférable de jouer sur la composition de composants ici que sur des conditions pour changer les labels et ajouter des règles spécifiques.

maxgfr commented on packages/code-du-travail-frontend/src/outils/RuptureCoventionnelle/steps/Resultat/utils.ts:
ça pour la rupture pro, on le garde ?

on avait la problématique avec le cdi d'opération également

m-maillot replied:
je n'ai pas compris ce que tu voulais dire.

maxgfr commented on packages/code-du-travail-frontend/src/outils/types.ts:
```diff
@@ -1,20 +1,21 @@
 
-export enum ToolName {
-  INDEMNITE_LICENCIEMENT = "Indemnité de licenciement",
+export enum IndemniteDepartType {
+  LICENCIEMENT = "Indemnité de licenciement",
+  RUPTURE_CONVENTIONNELLE = "Indemnité de rupture conventionnelle",
```
Ok je vois, pourquoi on a rebrand en type le toolName

maxgfr commented on packages/code-du-travail-modeles/src/modeles/base/__tests__/rupture-conventionnelle/calculate.spec.ts:
```diff
@@ -39,7 +39,7 @@ describe("Test de la fonctionnalité 'calculate'", () => {
     });
     expect(missingArgs[0].name).toEqual(
-      "contrat salarié . indemnité de licenciement . date d'entrée"
+      "contrat salarié - indemnité de licenciement - ancienneté en année"
```
Pourquoi avoir remis les `-`

maxgfr replied:
C'était mieux avec la logique de namespace avec les `.`

Viczei replied:
C'est vrai qu'il faudrait uniformiser

m-maillot replied:
En fait le comportement a changé ici. Avant on remontait en missingArgs manuellement (dans le ts) quand il manquait une des dates sur l'ancienneté. Sauf que quand on est sur la page info, on veut pouvoir faire un calculate sans passer de dates. On a donc changer sur l'indemnité de licenciement pour ne plus remonter des missingArgs depuis le TS. Ici c'est pareil. Du coup, le missingArgs est remonté par publicode où l'on a encore le replace des `.` par `-`. Pour le modifier, il faudrait basculer le préavis de retraitre pour ne plus utiliser hook form.

maxgfr reviewed (commented):
Top, énorme taff pour le refacto 💪

carolineBda committed:
feat(rupture co): refacto pour faire la comparaison du legal et du conventionnel côté modèle (#5734)

carolineBda committed:
Merge branch 'dev' into linked/feat/add-rupture-co

m-maillot committed:
feat: amélioration du calcul si plusieurs types de rupture

m-maillot committed:
fix: feedbacks

carolineBda committed:
fix: default choosen result (#5758)

m-maillot committed:
Merge branch 'dev' into linked/feat/add-rupture-co

maxgfr committed:
feat(rupture-co): ajout du support des CCs hors ANI (#5755)

* fix: done

* fix: done

carolineBda committed:
Merge branch 'dev' into linked/feat/add-rupture-co

maxgfr committed:
Merge branch 'dev' into linked/feat/add-rupture-co

maxgfr committed:
fix: build encore

maxgfr committed:
feat(rupture-co): ajout de la CC 1501 ; 1996 ; 44 (#5756)

* fix: done

* fix: done

* fix: done

* fix: done

* fix: done

* fix: done

* Update packages/code-du-travail-modeles/src/modeles/conventions/1501_restauration_rapide/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/1501_restauration_rapide/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/1996_pharmacie/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/44_industries_chimiques/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/1996_pharmacie/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/44_industries_chimiques/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* fix: build encore

* fix(ani): done

* fix(ani): done

---------

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

maxgfr committed:
Merge branch 'dev' into linked/feat/add-rupture-co

carolineBda reviewed (dismissed):
Régression sur la page résultat quand on coche licenciement pour inaptitude :

![Screenshot from 2024-04-16 17-20-34](https://github.com/SocialGouv/code-du-travail-numerique/assets/4971715/dd4308f1-bcb3-47d3-aa3b-e4945296f520)

Viczei committed:
refactor: calculate indemnite licenciement rupture (#5773)

* refactor: indemnite-licenciement et rupture conv

* fix: tests

* merge dev

* fix: tests

* chore: clean

* chore: clena

* fix: test

* chore: build fix

* fix: formule erreur

* chore: rename test

* fix: licenciement indemnite test

* fix: error when inaptitude = true

* chore: clean

* refactor: getFormule

* chore: clean

* fix: tests

* feat: filtré les résultats valeur 0

---------

Co-authored-by: Victor <victor.zeinstra@gmail.com>

maxgfr committed:
feat(rupture-co): ajout de la CC 2098 ; 1486 (#5782)

* fix: done

* fix: done

* fix: done

* fix: done

* fix: done

* fix: done

* Update packages/code-du-travail-modeles/src/modeles/conventions/1501_restauration_rapide/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/1501_restauration_rapide/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/1996_pharmacie/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/44_industries_chimiques/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/1996_pharmacie/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* Update packages/code-du-travail-modeles/src/modeles/conventions/44_industries_chimiques/__tests__/rupture-conventionnelle/calculate.spec.ts

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

* fix: build encore

* fix(ani): done

* fix(ani): done

* fix(ani): done

* fix(ani): done

* fix(ani): done

* fix(e2e): set number of idcc for h2 in contributions

---------

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

maxgfr committed:
fix: merge

carolineBda committed:
feat(rupture-co): ajout de la 2216, 1702 et 2120 (#5759)

* feat(RC): implement 2216 and 1702

* ajout de la 2120

* fix: rebase

* fix: calculate

* fix: calculate

---------

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

maxgfr committed:
fix: test

maxgfr committed:
fix: unit test...

maxgfr committed:
Merge branch 'dev' into linked/feat/add-rupture-co

m-maillot committed:
feat(rupture-co): gestion de la convention collective 16 (#5780)

maxgfr committed:
feat(events): utilisation d'events différents entre la rupture-co et l'indemnite-licenciement (#5771)

* fix: merge

* fix: event

* fix: event

* fix: event

* fix: finito

* fix: finito

maxgfr committed:
fix: import

maxgfr committed:
Merge branch 'dev' into linked/feat/add-rupture-co

maxgfr committed:
feat(rupture-co): ajout d'un point d'aide à l'étape ancienneté (#5809)

maxgfr committed:
fix: disable other

Viczei committed:
feat: 5790 indemnit de rupture conventionnelle exclure les cdi doprations de la simulation pour la 1404 (#5805)

* feat: impolement no cdi operation RC

* refactor: move rule to yaml

---------

Co-authored-by: Victor <victor.zeinstra@gmail.com>

Viczei committed:
feat: update wordings step salary (#5804)

Co-authored-by: Victor <victor.zeinstra@gmail.com>

Viczei committed:
feat: 5788 indemnit de rupture conventionnelle message dingibilit en cas de cdd (#5799)

* feat: implement html for ineligibility

* feat: add specific eligibility case for 3239

* feat: add specific eligibility case for 3239

* chore: review

* chore: ts fix

* chore: update build

* chore: update TU

---------

Co-authored-by: Victor <victor.zeinstra@gmail.com>

Viczei committed:
feat: implement 675 rupture co (#5797)

* refactor: indemnite-licenciement et rupture conv

* fix: tests

* merge dev

* fix: tests

* chore: clean

* chore: clena

* fix: test

* chore: build fix

* feat: implement 675 rupture co

* fix: formule erreur

* chore: rename test

* fix: licenciement indemnite test

* fix: error when inaptitude = true

* chore: clean

* refactor: getFormule

* chore: TU

* chore: TU

* chore: clean

* fix: tests

* chore: clean

* chore: clean

* fix: TU

---------

Co-authored-by: Victor <victor.zeinstra@gmail.com>

m-maillot committed:
feat(rupture-co): changement du wording sur les questions (#5802)

* feat(rupture-co): changement du wording sur les questions

* feat(rupture-co): désactiver licenciement pour motif absence prolongée ou répétées dans la CC 3248

* feat(rupture-co): correction sur les tests de la CC 16

* feat(rupture-co): fix unit test

m-maillot committed:
feat(rupture-co): refacto sur le retour du calculate (#5812)

* feat(rupture-co): refacto sur le retour du calculate

* feat(rupture-co): fix unit test

m-maillot committed:
feat(rupture-co): ajout de l'icone pour la rupture conventionnelle (#5814)

Viczei committed:
feat: 5783 indemnité rupture conventionnelle test UI (#5810)

* feat: add TU RC

* cd '/home/victor/Workspace/dnum/code-du-travail-numerique'

* feat: add test 1404 3239 + fix

* chore: clean

* chore: update TU

---------

Co-authored-by: Victor <victor.zeinstra@gmail.com>

maxgfr committed:
feat(rupture-co): ajout d'un message lors de la selection d'une CC hors ANI (#5811)

* fix: merge

* fix: done

* fix: test

* fix: test

* fix: merge

* fix: disable other

* fix: disable other

* fix: disable other

* fix: disable other

* fix: snapshot

* fix: snapshot

* fix: snapshot

* fix: snapshot

* fix: snapshot

* fix: test

* fix: test

* Revert "fix: test"

This reverts commit d31819c91668178406ad679e4aa67946eb940566.

* Revert "fix: test"

This reverts commit d625544afa6e842f456a10d1a558436451b0b374.

* fix: unit test

* fix: test

* fix: test

---------

Co-authored-by: Martial Maillot <martial.maillot@gmail.com>

maxgfr committed:
fix: merge

maxgfr committed:
fix: jest perf

maxgfr committed:
fix: upgrade jest

maxgfr committed:
fix: upgrade jest

maxgfr committed:
fix: upgrade jest

maxgfr committed:
fix: upgrade jest

m-maillot committed:
fix: icon

m-maillot committed:
feat(rupture-co): corrections suite aux tests (#5827)

maxgfr committed:
fix(rupture-co): modification wording pour la CC 2120 (#5821)

maxgfr committed:
fix(modeles): optimisation de la taille de la lib exportée (#5822)

* fix: opti

* fix: jest config

* fix: types
